src/library/scala/collection/immutable/List.scala

```diff
@@ -615,7 +647,21 @@ object List extends StrictOptimizedSeqFactory[List] {
     case coll: List[B] => coll
     case _ if coll.knownSize == 0 => empty[B]
     case b: ListBuffer[B] => b.toList
-    case _ => ListBuffer.from(coll).toList
+    case _ =>
+      val iter = coll.iterator
+      if (iter.hasNext) {
+        val head = new ::[B](iter.next(), Nil)
```

Comment:
Sorry, I thought I had posted this comment a week back:

@szeiger do you prefer we initialize the tails of `::`'s as `null` or as `Nil`? Doesn't matter to me which one, but they should probably be consistent.